### PR TITLE
Week 5 examples

### PR DESCRIPTION
**Pull request by ExtremeSkepticist**

### CODE REVIEW

File: 05/Cat.h [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#pragma once
+
+#include "Animal.h"
+
+namespace samples
+{
+	class Cat : public Animal
+	{
+	public:
+		Cat(int age, char* name);
```
Comment (popekim) [inline 05/Cat.h:10]: const char *

File: 05/Cat2.cpp [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+#include "Cat2.h"
+
+namespace samples
+{
+	Cat2::Cat2(int age, char* name)
+		: mAge(age), mName(name)
+	{
+	}
+
+	char* Cat2::GetType() const
```
Comment (popekim) [inline 05/Cat2.cpp]: this should be const char* and it should be.. .static function

File: 05/Cat2.cpp [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+	{
+		return mAnimalType;
+	}
+
+	char* Cat2::mAnimalType = "Cat";
```
Comment (popekim) [inline 05/Cat2.cpp:25]: put this one at the top

File: 05/Cat2.h [hunk 4]
```diff
@@ -0,0 +1,18 @@
+#pragma once
+
+namespace samples
+{
+	class Cat2
+	{
+	public:
+		Cat2::Cat2(int age, char* name);
+
+		char* GetType() const;
```
Comment (popekim) [inline 05/Cat2.h]: static.. .return const char *

File: 05/Dog.cpp [hunk 5]
```diff
@@ -0,0 +1,14 @@
+#include "Dog.h"
+
+namespace samples
+{
+	Dog::Dog(int age, char* address)
+		: Animal(age), mHomeAddress(address)
+	{
+	}
+
+	const char* Dog::GetAddress() const
```
Comment (popekim) [inline 05/Dog.cpp]: static

File: 05/Dog.h [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+#include "Animal.h"
+
+namespace samples
+{
+	class Dog : public Animal
+	{
+	public:
+		Dog(int age, char* address);
+
+		const char* GetAddress() const;
```
Comment (popekim) [inline 05/Dog.h]: static

File: 05/Dog.h [hunk 7]
```diff
@@ -0,0 +1,17 @@
+#pragma once
+
+#include "Animal.h"
+
+namespace samples
+{
+	class Dog : public Animal
+	{
+	public:
+		Dog(int age, char* address);
```
Comment (popekim) [inline 05/Dog.h:10]: const char*

File: 05/InlineFunctionExample.cpp [hunk 8]
```diff
@@ -0,0 +1,9 @@
+#include "InlineFunctionExample.h"
+
+namespace samples
+{
+	void InlineFunctionExample()
+	{
+
```
Comment (popekim) [inline 05/InlineFunctionExample.cpp:7]: why is it empty? cuz we have nothing to show? lol

**Review by popekim — CHANGES_REQUESTED** [review]

File: 05/InlineFunctionExample.cpp (hunk 8 above)
Comment (popekim) [reply 05/InlineFunctionExample.cpp:7]: then just remove this function and file alltogether that's fine

File: 05/ObjectAddressSavingExample.cpp [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+	{
+		Tiger* tiger1 = new Tiger(5);
+		uint32_t* unit32Number = reinterpret_cast<uint32_t*>(tiger1);
+
+		cout << unit32Number << endl; // you would save this on disk
+
+		//then when you load it again
+		Tiger* tiger2 = reinterpret_cast<Tiger*>(unit32Number);
+		tiger2->PretendIAmAZibra();
+	}
```
Comment (popekim) [inline 05/ObjectAddressSavingExample.cpp]: add delete tiger1 in the end

File: 05/ObjectPointerCastingExample.cpp [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+		Animal* myPet2 = new Dog(2, "BCIT");
+	
+		Cat* myCat = static_cast<Cat*>(myPet1);
+		Dog* myDog1 = static_cast<Dog*>(myPet2);
+		Dog* myDog2 = static_cast<Dog*>(myPet1);
+
+		cout << "myCat's name : " << myCat->GetName() << endl;
+		cout << "myDog1's address :" << myDog1->GetAddress() << endl;
+		cout << "myDog2's address : " << myDog2->GetAddress() << endl;
+	}
```
Comment (popekim) [inline 05/ObjectPointerCastingExample.cpp]: delete myPet1
delete myPet2

File: 05/StaticMemberVariableExample.cpp [hunk 11]
```diff
@@ -0,0 +1,21 @@
...
+#include "Cat2.h"
+#include "StaticMemberVariableExample.h"
+
+using namespace std;
+
+namespace samples
+{
+	void StaticMemberVariableExample()
+	{
+		Cat2* myCat1 = new Cat2(2, "Lulu");
```
Comment (popekim) [inline 05/StaticMemberVariableExample.cpp:11]: don't use new.. just creating it on stack is fine

File: 05/Vector2.cpp [hunk 12]
```diff
@@ -0,0 +1,84 @@
...
+
+namespace samples
+{
+	Vector2::Vector2()
+		: Vector2(0, 0)
+	{
+	}
+
+	Vector2::Vector2(int x, int y)
+		: mX(x)
```
Comment (popekim) [inline 05/Vector2.cpp:11]: all inline function implementation should be in header file... after class declaration

File: 05/Vector.h [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+{
+	class Vector
+	{
+	public:
+		Vector();
+		Vector(int x, int y);
+
+		inline int GetX() const;
+		inline void SetX(int x);
+
```
Comment (popekim) [inline 05/Vector.h:13]: inline implmentaiton should be all in .h file

File: 05/Dog.h (hunk 6 above)
Comment (ExtremeSkepticist) [reply 05/Dog.h]: So you want me to make this function return mHomeAddress as a static variable?

**Review by popekim — CHANGES_REQUESTED** [review]

**Pull request merged**